data/crud/unified/bulkWrite-replaceOne-dots_and_dollars.json

```diff
@@ -120,80 +120,6 @@
         }
       ]
     },
-    {
-      "description": "Replacing document with top-level dotted key on pre-3.6 server yields server-side error",
```

Comment:
Recommend keeping any specification tests that are still present in the specifications repo:
https://github.com/mongodb/specifications/blob/82be6f2203ee35a6c40af36e5233b632ef402058/source/crud/tests/unified/bulkWrite-replaceOne-dots_and_dollars.yml#L61

Even though this test is no longer expected to run (due to `maxServerVersion` being lower than 3.6), the test files are intended to be exact copies of what is in the spec repo. That enables easy updating.

Reply:
It looks like _all_ the tests I removed are in the specifications repo. I think this ticket doesn't need any additional work to resolve so I can just abandon this PR.

Reply:
Non-spec tests could be removed. Some tests check wire version ([example](https://github.com/mongodb/mongo-cxx-driver/blob/1baf35c085e61225f1d644a333bfa507ce8221dd/src/mongocxx/test/collection.cpp#L2198-L2215)). Tests only running when the maxWireVersion <= 6 could be removed. Wire version 6 [corresponds to server 3.6](https://github.com/mongodb/specifications/tree/master/source/wireversion-featurelist.md).

Though I expect it is not strictly necessary to remove tests. I'd suggest trying to remove those. If there is unexpected difficulty, then I think it is OK to skip.

Reply:
That's the one test case that only runs for maxWireVersion <= 6, so simple enough to change!